### PR TITLE
Sjukratryggingar/initial setup of health insurance application

### PR DESCRIPTION
**Pull request by dd-sdang**
# Sjukratryggingar - Initial setup of health insurance application

## What

Added new library for health insurance application in Application System with initial structure and dummy template.

## Why

New application in the Application System for Health Insurance

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/application/templates/health-insurance/README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<!-- gitbook-ignore -->
+
+# Application for health insurance
+
+Application for health insurance when moving back to Iceland after living abroad.
```
Comment (arnifreyrs) [inline libs/application/templates/health-insurance/README.md:5]: i am wondering, does a general application for health insurance exist somewhere for Sjúkratryggingar?

My concern is that maybe this application you are working on is specific to health insurance for people moving back to iceland, so the HEALTH_INSURANCE enum value is "taken".

File: libs/application/templates/health-insurance/src/lib/HealthInsuranceTemplate.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  infoInput: z.string(),
+  summaryInput: z.string(),
+})
+
+const HealthInsuranceTemplate: ApplicationTemplate<
+  ApplicationContext,
+  ApplicationStateSchema<Events>,
+  Events
+> = {
+  type: ApplicationTypes.EXAMPLE,
```
Comment (arnifreyrs) [inline libs/application/templates/health-insurance/src/lib/HealthInsuranceTemplate.ts:28]: here you have to use the `ApplicationTypes.HEALTH_INSURANCE`

Comment (dd-sdang) [reply libs/application/templates/health-insurance/src/lib/HealthInsuranceTemplate.ts]: Great you spotted it, had missed this. Updated it 👍

**Review by darrikonn — COMMENTED** [review]
This is missing the babel config, most likely not set up as a react lib.

File: libs/application/templates/health-insurance/README.md (hunk 1 above)
Comment (dd-sdang) [reply libs/application/templates/health-insurance/README.md:5]: I have investigated it and it seems that this is the general application. If an Icelandic citizen never moves abroad they will get health insurance automatically and doesn't need to apply for it. But I could always make this enum specific anyway if we want to be safe. What do you think?

For more background, there will be applications in the future for insurance claim from the Health insurance for example sick pay and work accidents. But that might not be conflicting with this enum as they are more specific.

Comment (darrikonn) [reply libs/application/templates/health-insurance/README.md:5]: Good comment ^

**Comment by dd-sdang** [comment]
> This is missing the babel config, most likely not set up as a react lib.

@darrikonn Oh no, i must have taken the wrong command and used the node library generation instead. How do I update this in the best way? Should I regenerate it?

**Comment by darrikonn** [comment]
> > This is missing the babel config, most likely not set up as a react lib.
> 
> @darrikonn Oh no, i must have taken the wrong command and used the node library generation instead. How do I update this in the best way? Should I regenerate it?

You can just copy-paste the `tsconfig.*` + `(.?)babel*` + `.eslintrc` files from the reference template to your template.

File: libs/application/templates/health-insurance/README.md (hunk 1 above)
Comment (arnifreyrs) [reply libs/application/templates/health-insurance/README.md:5]: great answer, I think that this enum value (HEALTH_INSURANCE) can safely be reserved for this application.

**Review by arnifreyrs — APPROVED** [review]

**Review by darrikonn — APPROVED** [review]

**Pull request merged**